packages/axios-extension/src/abap/ui5-abap-repository-service.ts

```diff
@@ -158,12 +162,20 @@ export class Ui5AbapRepositoryService extends ODataService {
             if (response?.headers?.['sap-message']) {
                 prettyPrintMessage({ msg: response.headers['sap-message'], log: this.log, host: frontendUrl });
             }
-            // log url of created/updated app
-            const path = '/sap/bc/ui5_ui5' + (!bsp.name.startsWith('/') ? '/sap/' : '') + bsp.name.toLowerCase();
-            const query = this.defaults.params?.['sap-client']
-                ? '?sap-client=' + this.defaults.params['sap-client']
-                : '';
-            this.log.info(`App available at ${frontendUrl}${path}${query}`);
+            if (!testMode) {
+                // log url of created/updated app
+                const path = '/sap/bc/ui5_ui5' + (!bsp.name.startsWith('/') ? '/sap/' : '') + bsp.name.toLowerCase();
+                const query = this.defaults.params?.['sap-client']
+                    ? '?sap-client=' + this.defaults.params['sap-client']
+                    : '';
+                this.log.info(`App available at ${frontendUrl}${path}${query}`);
+            } else if (testMode) {
```

Comment:
the 2nd if is redundant (i.e no need to check `testmode` again, it must be truthy). 
what does prettyPrintError do if there is no error?

Reply:
Please refer to the 2nd screenshot above, it prints the error nodes returned.